test/Constraints/optional.swift

```diff
@@ -565,3 +565,16 @@ func testFunctionContainerMethodCall(container: FunctionContainer?) {
   // expected-note@-3 {{coalesce}}
   // expected-note@-4 {{force-unwrap}}
 }
+
+// Test for https://github.com/apple/swift/issues/60730
+// rdar://94037733
+do {
+  struct S: P {}
+  func takesP(_: any P) {}
+  func passOptional(value: (any P)?) {
+    takesP(value)
+    // expected-error@-1 {{value of optional type '(any P)?' must be unwrapped to a value of type 'any P'}}
+    // expected-note@-2 {{coalesce using '??' to provide a default when the optional value contains 'nil'}}
+    // expected-note@-3 {{force-unwrap using '!' to abort execution if the optional value contains 'nil'}}
+  }
```

Comment:
@dfperry5 Could you please add one more test-case that tests the opposite scenario where argument does not conform to `P` to make sure that we don't suggest the unwrap? it could be i.e. `(any BinaryInteger)?`) or any stdlib protocol so you don't have to declare anything new.

Reply:
@xedin since we are here, I also think would make sense a test case with more than one layer of optional like `(any P)??`

Reply:
@xedin  / @LucianoPAlmeida  -- added these tests.

One thing - wondering if I made a mistake somewhere, for the optional layers, the error message is: 

" error: value of optional type '(any P)??' must be unwrapped to a value of type '(any P)?' "

Is that right? I was wondering if it should be:

" error: value of optional type '(any P)??' must be unwrapped to a value of type 'any P' "